### DISCUSSION TITLE
[Idea] Linux Support for Bitdefender Antivirus

### DISCUSSION THREAD

**Discussion by OscarBeiro** [glpi-project/glpi-agent#470] [Ideas]
### Your idea

Inspired by #458 we've tried to add support for Bitdefender Endpoint Security Tools, since it is our antimalware solution.

The command is bduitool but doesn't return a json.

Command:
```
/opt/bitdefender-security-tools/bin/bduitool get ps
```
Content returned:
```
Product version: 7.0.3.2239
Last product update: 2023-08-24 at 10:44:23 GMT+2
New product update available: no
Engines version: 7.95171
Last security content update: 2023-08-24 at 11:44:37 GMT+2
New security content available: no
Installed scan type: Full
Installed scan type fallback: None
Currently used scan type: Full
Features:
  - Antimalware status: On
  - NetworkMonitor status: On
```

Product version and Engines version are assigned directly
- Antimalware status: 0n should be  the choice for setting the active to Yes
- If New product update available and New security content available are set to no, Up to date should be YES.
I don't think we can retrieve any other info since GravityZone handles everything. 
In fact the Windows solutions returns a lot less info.

Finally we have even started to make some substitution to the defender code, but it is not working. Having no clue about Perl might be a reason! :)

I will create a PR to see if anybody can help

**Comment by OscarBeiro** [comment]
PR https://github.com/glpi-project/glpi-agent/pull/472

**Reply by g-bougard** [reply]
Hi @OscarBeiro 

thank you for your PR, I just reviewed it.

I'm curious about one point, not knowing this solution. Is this a free antivirus ? Or can it requires to pay a license in some cases ? This question to know if the "expiration" value required to be set some times.

**Reply by OscarBeiro** [reply]
Hi, 
Sorry about the delay. I am on vacation (allegedly).

It's not a free solution, unfortunately. In my experience, having an updated security tool for free is nearly impossible because of the effort needed.
https://www.bitdefender.com/business/support/en/77209-77513-linux-agent.html

Regarding your expiration question, it will be great to have, but since it is cloud-managed, the information regarding licensing is not in the client.
Believe it or not, the Windows client has much less information than the Linux one. 

So, there's not a way to fill it, AFAIK.
I'm testing your suggestions and will answer on the PR.

Merci.

**Reply by OscarBeiro** [reply]
Also, I would like to point that this solution blocked the October attacks on our vulnerable GLPIs, so, it is good to know its status from the GLPI itself. :)

**Reply by g-bougard** [reply]
Hello,

thank you for the answers.

I checked quickly the site from bitdefender. I saw there's a way to obtain the expiration date, but this is through the public API:
https://www.bitdefender.com/business/support/en/77209-127105-getlicenseinfo.html

I really checked it quickly, but as the API seems to also permit license key removing or sensible information leaks, this is probably a bad idea to provide an [API key](https://www.bitdefender.com/business/support/en/77209-125277-public-api.html) to the agent. But I think you can provide a script to catch that information so it is less easy to discover the precious API key for anybody who may have gain unwanted access to the computer:
1. it may have to be run daily (or less often) from crontab to put the result in a place the module will read it for integration in inventory, or it can create a json to be used with --additional-content option,
2. or it may just be a command to put in the root PATH or in a specific full path which returns the expected date and you then just have to integrate that command support in the module

The choice is up to you. And of course, the chosen process would have to be documented somewhere in public GLPI Agent documentation. But I can handle the documentation part if you want.

**Reply by OscarBeiro** [reply]
Hi,

You are right. As they are now, API permissions don't allow a read-only key. It seems a lot of work for one particular date, but I will look into it once we have some information from Bitdefender in GLPI. :)

Merci

**Reply by OscarBeiro** [reply]
Hi,
I have reviewed the getLicenseInfo endpoint and I think it is overkill to use it.
We are managing licenses for several clients, and the key will have access to all that information. In my opinion, dropping the APIKey on another guy's computer is not a good idea.
Also, we will need to find a way to match each license with the installed client, and I don't think there's an easy way for that.

So, I think the Gravityzone will remain on my lengthy wishlist of future GLPI integrations

**Reply by g-bougard** [reply]
Okay, maybe this a place for a dedicated GLPI plugin ;-)

**Reply by g-bougard** [reply]
Hello @OscarBeiro 

as I'm adding an unittest with your bduitool output, I'm having a question: you decided to set `Engines version` as `BASE_VERSION`. From a semantic view, this doesn't seem to match the purpose. In GLPI this is reported as `Signature database version` as [your screenshot here shows](https://github.com/glpi-project/glpi-agent/pull/472#issuecomment-1706911579). Is `Engines version` really updated each time the Bitdefender database is ?

**Reply by OscarBeiro** [reply]
I understand is weird, but it is what it makes more sense to me.
This is my current result:
```
Product version: 7.0.3.2239
Last product update: 2023-09-05 at 19:04:47 GMT+2
New product update available: yes
Engines version: 7.95234
Last security content update: 2023-09-07 at 17:14:26 GMT+2
New security content available: no
Installed scan type: Full
Installed scan type fallback: None
Currently used scan type: Full
Features:
  - Antimalware status: On
  - NetworkMonitor status: Off

```
So, Engines version is updated continuously it is what it makes sense to me to equal signature database. Product update is 2 days old, which is a lot for a security solution.

Regards,
